packages/aws-cdk-lib/aws-apigateway/lib/method.ts

```diff
@@ -199,6 +199,12 @@ export class Method extends Resource {
         `which is different from what is required by the authorizer [${authorizer.authorizationType}]`);
     }
 
+    // When AuthorizationType is None, there shouldn't be any AuthorizationScope since AuthorizationScope should only
+    // be applied to COGNITO_USER_POOLS AuthorizationType.
+    const defaultScopes = options.authorizationScopes ?? defaultMethodOptions.authorizationScopes;
+    const authorizationScopes = authorizationTypeOption === AuthorizationType.COGNITO ? defaultScopes : undefined;
+    Annotations.of(this).addWarningV2('@aws-cdk/aws-apigateway:invalidAuthScope', '\'AuthorizationScopes\' can only be set when \'AuthorizationType\' sets \'COGNITO_USER_POOLS\'. Default to ignore the values set in \'AuthorizationScopes\'.');
```

Comment:
Why do we warn instead of throwing an error when AuthorizationType isn't Congnito but AuthorizationScopes is set?

Reply:
We want to warn here because there's a subtle use case. When a API creates a OPTIONS method in addition to the POST/GET method (i.e. when authorizor is used), the POST/GET method is having the correct behaviour. However, the OPTIONS method will set auth scope regardless of auth type. The OPTIONS method is not explicitly managed by users. If we raise exception, then users would not be able to use the implicitly created OPTIONS method along with authorizor. Instead warn and continue seems like a better approach.

Reply:
mmm, i see. Thanks for the explanation